### ISSUE TITLE
[MCU] Communication Layer (IsoTP)

### ISSUE THREAD

**Issue by AndreasFreund** [amosproj/amos2024ss07-updating-flash-boot-loader#40]
## User story
1. As a developer
2. I need a module to send and receive IsoTP messages
3. So that I can send messages bigger than normal 8 bytes length

## Acceptance criteria
* Reassemble packets on reception
* Demonstrating by sending example packets (one short and one long) on CAN bus
* Demonstrate reception of messages sent (e.g. by other devkit, if you have another one available or by PC)
* Take care of limited RAM of MCU, e.g. by using fixed size buffer

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
